### PR TITLE
Support in-config data connection string

### PR DESCRIPTION
**Pull request by mamaso**

### CODE REVIEW

**Comment by mamaso** [comment]
resolves #238

File: src/configuration/fromEnvironment.js [hunk 1]
```diff
@@ -36,7 +37,7 @@ module.exports = function (configuration, environment) {
             case 'sqlconnstr_ms_tableconnectionstring':
             case 'sqlazureconnstr_ms_tableconnectionstring':
             case 'ms_tableconnectionstring':
-                configuration.data = assign(configuration.data || {}, connectionString.parse(environment[key]));
+                configuration.data.connectionString = environment[key];
```
Comment (danderson00) [inline src/configuration/fromEnvironment.js]: This is much cleaner!

File: src/index.js [hunk 2]
```diff
@@ -67,6 +68,8 @@ module.exports = function (configuration, environment) {
     loadConfiguration.fromEnvironment(configuration, environment || process.env);
     loadConfiguration.fromSettingsJson(configuration);
 
+    configuration.data = assign(configuration.data, connectionString.parse(configuration.data.connectionString));
```
Comment (danderson00) [inline src/index.js]: *should* be able remove the LHS of this expression, i.e. remove "configuration.data = ". assign should assign values to the object supplied as the first argument. (should being the keyword here)

Comment (mamaso) [inline src/index.js]: alright, I'll make that change & see if the tests still pass

Comment (mamaso) [inline src/index.js]: Looks like the assignment is necessary, deeply just returns a cloned/merged object and doesn't do any assignment

Comment (danderson00) [inline src/index.js]: Humm.... Not according to the deeply spec - https://www.npmjs.com/package/deeply.

Comment (mamaso) [inline src/index.js]: I looked at the spec earlier, it isn't the clearest thing in the world.  Notice how throughout there are two temporary objects being merged with no assignment, which doesn't make much sense.

If you actually saw words to that ^^ effect then something else may be going on

Comment (mamaso) [inline src/index.js]: Yep, did another test, it just returns the merged obj

Comment (danderson00) [inline src/index.js]: Indeed. In this case, deeply is really a merge, not an assign. I guess we keep the LHS! No wuccas.

**Pull request merged**